packages/core/status/core-status-server-internal/src/routes/status.ts

```diff
@@ -53,19 +53,17 @@ interface StatusHttpBody extends Omit<StatusResponse, 'status'> {
   status: StatusInfo | LegacyStatusInfo;
 }
 
-export interface RedactedStatusHttpBody {
-  status: {
-    overall: {
-      level: ServiceStatusLevel;
-    };
-  };
-}
-
 const SERVICE_UNAVAILABLE_NOT_REPORTED: ServiceStatus = {
   level: ServiceStatusLevels.unavailable,
   summary: 'Status not yet reported',
 };
 
+const responseSchema = schema.oneOf([redactedStatusResponse, statusResponse], {
+  meta: {
+    description: `Kibana's operational status. A minimal response is sent for unauthorized users.`,
+  },
+});
```

Comment:
So, my concern is, we're basically generating schemas (impacting both memory footprint and start time), that will only be used **if** OAS generation is enabled.

So my question is: Can we find a pattern so that those schemas are only generated **if** OAS generation is enabled?

Just one schema is fine, but if we go that for all our routes (and I assume we want to?), this could be impactful. 

WDYT?

Reply:
Resolved in https://github.com/elastic/kibana/pull/181622